sdk/dotnet/Helm/ChartBase.cs

```diff
@@ -111,15 +111,8 @@ protected ChartBase(string releaseName, Union<ChartArgs, LocalChartArgs> args, C
                         flags.Add(cfgBase.Namespace);
                     }
 
-                    // Use the HELM_HOME environment variable value if set.
-                    var home = Environment.GetEnvironmentVariable("HELM_HOME");
-                    if (home != null)
-                    {
-                        flags.Add("--home");
-                        flags.Add(home);
-                    }
-
-                    var yaml = Utilities.ExecuteCommand("helm", flags.ToArray());
+                    var yaml = Utilities.ExecuteCommand("helm", flags.ToArray(),
+                        (Dictionary<string, string>)Environment.GetEnvironmentVariables());
```

Comment:
@mikhailshilkov CI is failing with the following error:
```
kubernetes:helm.sh:Chart empty1 creating error: Pulumi.ResourceException: Unable to cast object of type 'System.Collections.Hashtable' to type 'System.Collections.Generic.Dictionary`2[System.String,System.String]'
```

I'm guessing this line is at fault, but I'm not really sure how to fix. Any thoughts?

Reply:
Unfortunately, `Environment.GetEnvironmentVariables` is a shitty old method that nobody touched for 15 years, so it returns some ancient types. It won't convert to a generic dictionary OOTB.

We likely want a helper wrapper, e.g. in `Utilities`, something like

```cs
public static IDictionary<string, string> GetEnvironmentVariables()
{
    var result = new Dictionary<string, string>();
    foreach (var variable in Environment.GetEnvironmentVariables())
    {
        if (variable == null) continue;

        var entry = (DictionaryEntry)variable; 
        if (entry.Key is string key && entry.Value is string value)
            result.Add(key, value);
    }
    return result;
}
```

It's s chunk of defensive code, but it's kind of needed to make the compiler happy.